include/inexor/vulkan-renderer/world/cube.hpp

```diff
@@ -75,7 +75,7 @@ class Cube : public std::enable_shared_from_this<Cube> {
     void remove_childs();
 
     /// Get the root to this cube.
-    [[nodiscard]] std::weak_ptr<Cube> root() const noexcept;
+    [[nodiscard]] std::shared_ptr<Cube> root() noexcept;
```

Comment:
Can this still be a const method?

Reply:
No as we want to return a `shared_ptr<Cube>` and thats not possible with `const` and `shared_from_this()`

Reply:
Oh right.